src/context.cc

```diff
@@ -2839,23 +2839,10 @@ error Context::DeleteTimeEntryByGUID(const std::string GUID) {
 error Context::SetTimeEntryDuration(
     const std::string GUID,
     const std::string duration) {
-    if (GUID.empty()) {
-        return displayError(std::string(__FUNCTION__) + ": Missing GUID");
-    }
-
-    Poco::Mutex::ScopedLock lock(user_m_);
-    if (!user_) {
-        logger().warning("Cannot set duration, user logged out");
-        return noError;
-    }
-    TimeEntry *te = user_->related.TimeEntryByGUID(GUID);
-    if (!te) {
-        logger().warning("Time entry not found: " + GUID);
-        return noError;
-    }
-
-    if (isTimeEntryLocked(te)) {
-        return logAndDisplayUserTriedEditingLockedEntry();
+    TimeEntry *te = nullptr;
+    error findError = GetTimeEntryFromGUID(GUID, &te);
+    if (findError != noError && !te) {
```

Comment:
@NghiaTranUIT Can you explain why we add `&& !te` to the condition?
From my understanding, when there is some time entry but it is locked, we should not continue and should just return the error.

Reply:
If we take a look the original implementation (in master ) 

https://github.com/toggl/toggldesktop/blob/4d662731dc10ccb91189e2a71918f141f8ae3009/src/context.cc#L2793-L2796

You can see that if `!user_` is null, the func will return `noError` immediately.

So, `!te` is make sure that the TimeEntry is not null, before proceeding.

Reply:
Maybe you meant `|| !te` instead of `&& !te`?

Reply:
wow, you're right 😮 My brain 🧠 is gone